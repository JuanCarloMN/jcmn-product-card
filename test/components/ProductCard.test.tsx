import React from "react";
import renderer from 'react-test-renderer';
import { ProductCard } from "../../src/components/ProductCard";
import { product1 } from "../data/products";

// const { act } = renderer;

describe('ProductCard', () => {

    test('Debe de mostrar el compoente correctamente', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCard>
        ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de incrementar el contador', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{ count }</span>
                            <button onClick={ () => increaseBy(1) }></button>
                        </>
                    )
                }
            </ProductCard>
        ).toJSON();

        let tree = wrapper;
        expect( tree ).toMatchSnapshot();
        
        // act( () => {
        //     (tree as any).children[2].this.props.onClick();
        // })

        // tree = wrapper;

        // expect( (tree as any).chidren[1].children[0] ).toBe('1');
    })
    
})
