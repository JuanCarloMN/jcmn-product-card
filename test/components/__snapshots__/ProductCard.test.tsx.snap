// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe de incrementar el contador 1`] = `null`;

exports[`ProductCard Debe de mostrar el compoente correctamente 1`] = `null`;
